cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -379,6 +380,16 @@ List<ProgramHistory> getRuns(Collection<ProgramReference> programs, ProgramRunSt
   List<ProgramSpecification> getDeletedProgramSpecifications(ApplicationReference appRef,
                                                              ApplicationSpecification specification);
 
+  /**
+   * updates source control meta of an application
```

Comment:
Updates source control metadata for one or more applications

Reply:
Like for other batch APIs, this should clearly describe the expected behavior in relevant edge cases. For example, what is the expected behavior if one of the applications in the give collection does not exist.